app/controllers/discourse_ai/ai_helper/assistant_controller.rb

```diff
@@ -53,19 +53,32 @@ def suggest
       end
 
       def suggest_title
-        input = get_text_param!
+        assistant = DiscourseAi::AiHelper::Assistant.new
+
+        if params[:topic_id]
+          topic = Topic.find_by(id: params[:topic_id])
+
+          topic_content =
+            topic
+              .posts
```

Comment:
In the summarization lib we have some logic to deal with mega topics that may be useful here to adopt or reuse.

Like if the topic has 10000 replies, we take the best 50 and so on...